src/v/kafka/server/handlers/list_partition_reassignments.cc

```diff
@@ -0,0 +1,135 @@
+
+// Copyright 2020 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#include "kafka/server/handlers/list_partition_reassignments.h"
+
+#include "cluster/metadata_cache.h"
+#include "cluster/topics_frontend.h"
+#include "cluster/types.h"
+#include "kafka/protocol/errors.h"
+#include "kafka/protocol/schemata/list_partition_reassignments_request.h"
+#include "kafka/protocol/schemata/list_partition_reassignments_response.h"
+#include "kafka/server/errors.h"
+#include "kafka/server/fwd.h"
+#include "kafka/server/quota_manager.h"
+#include "model/metadata.h"
+#include "model/namespace.h"
+#include "model/timeout_clock.h"
+
+#include <seastar/core/coroutine.hh>
+#include <seastar/core/sstring.hh>
+
+#include <absl/container/node_hash_map.h>
+
+#include <algorithm>
+#include <chrono>
+#include <iterator>
+#include <vector>
+
+namespace kafka {
+
+template<>
+ss::future<response_ptr> list_partition_reassignments_handler::handle(
+  request_context ctx, [[maybe_unused]] ss::smp_service_group ssg) {
+    list_partition_reassignments_request request;
+    request.decode(ctx.reader(), ctx.header().version);
+    log_request(ctx.header(), request);
+    list_partition_reassignments_response resp;
+
+    if (!request.data.topics.has_value() || request.data.topics->empty()) {
```

Comment:
Can be omitted

Reply:
A check for `has_value` or something like that is necessary because `topics` is an `optional<vector>`.
I'll change the for-loop below to use `optional.value_or(empty vector)`